IDinsight/mosaiks
# Unsorted stac search

## Changes

### 1. Removed `convex_hull` from `fetch_stac_items`
Found that we don’t need to take the “[convex_hull](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.convex_hull.html)” of the batched union of the points (which essentially makes a polygon which fully encapsulates all the points) when fetching STAC references. A simple union works fine and means we don’t find STAC items that don’t actually overlap any points and just happen to overlap with encapsulating geometry somewhere random.

### 2. Made sorting the coords set into a config parameter in `featurisation.yaml`
- Since we're not doing convex hull anymore it means that we don't HAVE to sort the points spatially first before splitting into batches. It still is much faster to do this sorting because the set of overlapping images found would be smaller if the points are closer together, so the default should still be `sort_points = True`.

## To test
- Run the featurisation notebook to make sure everything works.
- Run through `stac_search_experiments.ipynb` and experiment with different point sets to see speed differences when points are closeby or far apart. Also test adding `.convex_hull` back to the code and seeing the difference (as described in notebook)

# To do before merging
- Merge PR #112 first.
